fluentgen/src/main/java/com/azure/autorest/fluent/mapper/FluentLiveTestsMapper.java

```diff
@@ -0,0 +1,154 @@
+import com.azure.autorest.fluent.model.clientmodel.examplemodel.FluentResourceUpdateExample;
+import com.azure.autorest.fluent.model.clientmodel.fluentmodel.create.ResourceCreate;
+import com.azure.autorest.fluent.model.clientmodel.fluentmodel.update.ResourceUpdate;
+import com.azure.autorest.fluent.template.FluentExampleTemplate;
+import com.azure.autorest.fluent.util.FluentJavaSettings;
+import com.azure.autorest.fluent.util.FluentUtils;
+import com.azure.autorest.fluent.util.Utils;
+import com.azure.autorest.model.clientmodel.ExampleLiveTestStep;
+import com.azure.autorest.model.clientmodel.LiveTestStep;
+import com.azure.autorest.model.clientmodel.LiveTests;
+import com.azure.autorest.model.clientmodel.ProxyMethodExample;
+import com.azure.autorest.util.CodeNamer;
+
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A mapper to map vanilla live tests to fluent live tests.
+ */
+public class FluentLiveTestsMapper {
+    private final PluginLogger logger = new PluginLogger(FluentGen.getPluginInstance(), FluentLiveTestsMapper.class);
+
+    private final FluentExampleTemplate fluentExampleTemplate = FluentExampleTemplate.getInstance();
+
+    private static final FluentLiveTestsMapper INSTANCE = new FluentLiveTestsMapper();
+
+    public static FluentLiveTestsMapper getInstance(){
+        return INSTANCE;
+    }
+
+    public FluentLiveTests map(LiveTests liveTests, FluentClient fluentClient, CodeModel codeModel, FluentJavaSettings fluentJavaSettings) {
+
+        FluentLiveTests.Builder resultBuilder = FluentLiveTests.newBuilder();
+
+        resultBuilder.className(liveTests.getFilename() + "Tests");
+
+        resultBuilder.addTestCases(liveTests.getTestCases().stream().map(liveTestCase -> {
+            FluentLiveTestCase.Builder testCaseBuilder = FluentLiveTestCase.newBuilder().methodName(CodeNamer.toCamelCase(liveTestCase.getName()));
+            testCaseBuilder.addSteps(
+                liveTestCase.getTestSteps()
+                    .stream()
+                    // future work: support other step types
+                    .filter(testStep -> testStep instanceof ExampleLiveTestStep)
+                    .map((Function<LiveTestStep, Optional<FluentLiveTestStep>>) step -> {
+                        ExampleLiveTestStep exampleStep = (ExampleLiveTestStep) step;
+                        String operationId = exampleStep.getOperationId();
+                        String[] oprs = getOperationGroupPair(operationId, codeModel, fluentJavaSettings);
+                        String operationGroup = oprs[0];
+                        String operation = oprs[1];
+                        ProxyMethodExample example = exampleStep.getExample();
+                        FluentResourceCollection resourceCollection = findResourceCollection(fluentClient, operationGroup);
+
+                        FluentExampleTemplate.ExampleMethod exampleMethod = null;
```

Comment:
We probably need some refactor about these. Ideally Mapper should not refer Template.

Log an issue and consider this in next PR.

Reply:
Agreed, definitely needs refactoring... Logged.